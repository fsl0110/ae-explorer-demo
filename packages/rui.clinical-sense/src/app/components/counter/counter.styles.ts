import styled from "styled-components";

// basic exaple for styling html tags
export const CounterStyles = styled.div`
  background: #000;
  color: #fff;
`;
