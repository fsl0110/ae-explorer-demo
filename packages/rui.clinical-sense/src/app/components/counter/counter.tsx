import React, {FC} from "react";
import {Button} from "antd";
import {CounterStyles} from "./counter.styles";

interface Props {
  value: any;
  handleIncrement: () => void;
  handleDecrement: () => void;
  handleUndo: () => void;
  handleRedo: () => void;
}

export const Counter: FC<Props> = ({
  value,
  handleIncrement,
  handleDecrement,
  handleUndo,
  handleRedo,
}) => (
  <CounterStyles>
    Clicked: {value} times <Button onClick={handleIncrement}>+</Button>&nbsp;
    <Button onClick={handleDecrement}>-</Button>&nbsp;
    <Button onClick={handleUndo}>undo</Button>&nbsp;
    <Button onClick={handleRedo}>redo</Button>
  </CounterStyles>
);
