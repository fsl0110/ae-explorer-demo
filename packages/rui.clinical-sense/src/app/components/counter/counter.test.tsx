import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {Counter} from "./counter";

describe("<Counter>", () => {
  it("Snapshot", () => {
    expect(
      toJson(
        shallow(
          /* tslint:disable-next-line */
          <Counter
            value={1}
            handleIncrement={jest.fn()}
            handleDecrement={jest.fn()}
            handleUndo={jest.fn()}
            handleRedo={jest.fn()}
          />,
        ),
      ),
    ).toMatchSnapshot();
  });
});
