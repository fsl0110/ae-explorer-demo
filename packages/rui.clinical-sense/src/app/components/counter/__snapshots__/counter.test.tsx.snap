// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter> Snapshot 1`] = `
<styled.div>
  Clicked: 
  1
   times 
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[MockFunction]}
  >
    +
  </Button>
   
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[MockFunction]}
  >
    -
  </Button>
   
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[MockFunction]}
  >
    undo
  </Button>
   
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[MockFunction]}
  >
    redo
  </Button>
</styled.div>
`;
